POC/api/main.py

```diff
@@ -133,19 +165,250 @@ async def create_database(form: Annotated[DbInfoForm, fastui_form(DbInfoForm)]):
+        dbs: Sequence[DbInfo] = session.exec(statement).all()
+
+    return dbs
+
+
+@app.get(
+    "/api/forms/update/database/{database_id}",
+    response_model=FastUI,
+    response_model_exclude_none=True,
+    include_in_schema=False,
+)
+async def get_update_database_form(database_id: int):
+    with Session(db_engine) as session:
+        statement = select(DbInfo).where(DbInfo.id == database_id)
+        db: DbInfo | None = session.exec(statement).first()
+        if db is None:
+            raise HTTPException(status_code=404, detail="Database not found")
+        if db is not None:
+            db_info: DbInfoForm = DbInfoForm(**db.model_dump())
+
+    return [
+        c.Page(
+            components=[
+                c.Div(
+                    components=[
+                        c.Markdown(text="---"),
+                        c.ModelForm(
+                            submit_url=f"/api/forms/update/database/{database_id}",
+                            initial=db_info.model_dump(),
+                            loading=[c.Text(text="Submitting")],
+                            model=DbInfoForm,
+                        ),
+                    ]
+                ),
+                c.Markdown(text="---"),
+            ]
+        )
+    ]
+
+
+@app.post(
+    "/api/forms/update/database/{database_id}",
+    response_model=FastUI,
+    response_model_exclude_none=True,
+    include_in_schema=False,
+)
+async def submit_update_database_form(
+    database_id: int, form: Annotated[DbInfoForm, fastui_form(DbInfoForm)]
+):
+    with Session(db_engine) as session:
+        statement = select(DbInfo).where(DbInfo.id == database_id)
+        db: DbInfo | None = session.exec(statement).first()
+        if db is None:
+            raise HTTPException(status_code=404, detail="Database not found")
+        db.short_name = form.short_name
+        db.display_name = form.display_name
+        db.category = form.category
+        db.alias = form.alias
+        db.description = form.description
+        db.updated_at = dt.now()
+        session.add(db)
+        session.commit()
+        session.refresh(db)
+    return [
+        c.Page(
+            components=[
+                c.Div(
+                    components=[
+                        c.Heading(text=f"Updated: {db.name}", level=3),
+                        c.Table(data=[db]),
+                    ]
+                ),
+            ]
+        )
+    ]
+
+
+@app.put(
+    "/api/backend/update/database/{database_id}",
+    response_model=DbInfo,
+    response_model_exclude_none=True,
+    tags=["Database"],
+)
+async def api_update_database(database_id: int, db: DbInfoForm):
+    with Session(db_engine) as session:
+        statement = select(DbInfo).where(DbInfo.id == database_id)
+        old_db: DbInfo | None = session.exec(statement).first()
+        if old_db is None:
+            raise HTTPException(status_code=404, detail="Database not found")
+        old_db.name = db.name
```

Comment:
is there a better way to update the model? or is this correct?

i could update it via **model_dump of the model passed in but i'm not sure if thats slower even though it's cleaner.

Pydantic is currently working on a FastModel which all of the data stays in rust unless model_dump is called.